SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -1319,14 +1322,16 @@ public async Task<IActionResult> YourRecommendations_Post(YourRecommendationsVie
         [HttpPost("your-recommendations-download/{reference}")]
         public async Task<IActionResult> GenerateRecommendationsPdf_Post(string reference)
         {
-            var stream = await pdfGenerationService.GeneratePdf($"energy-efficiency/pdf-generation/your-recommendations/{reference}");
+            var encodedReference = HttpUtility.HtmlEncode(reference);
```

Comment:
I've just realised now this is the wrong thing to do - since this is going to be part of a url we should be calling the `UrlEncode` method. It's on the same class so a tiny fix.

Reply:
Good catch 🤦🏾‍♂️